### PR TITLE
Global refactor code

### PR DESCRIPTION
**Pull request by osmovia**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,47 @@
...
-/.idea/assetWizardSettings.xml
+# Mac OS X files
 .DS_Store
-/build
-/captures
-.externalNativeBuild
-.cxx
+
+# Android Studio files
+.idea
```
Comment (skliarov) [inline .gitignore:5]: Good!

File: app/src/main/java/com/example/learingrealmandretrofit/SwipeToDeleteCard.kt [hunk 2]
```diff
@@ -75,23 +49,10 @@ abstract class SwipeToDeleteCard(context: Context) :
         val deleteIconBottom = deleteIconTop + intrinsicHeight
 
         // Draw the delete icon
-        deleteIcon!!.setBounds(
-            deleteIconLeft,
-            deleteIconTop,
-            deleteIconRight,
-            deleteIconBottom
-        )
+        deleteIcon!!.setBounds(deleteIconLeft, deleteIconTop, deleteIconRight, deleteIconBottom)
```
Comment (skliarov) [inline app/src/main/java/com/example/learingrealmandretrofit/SwipeToDeleteCard.kt:52]: Double exclamation marks

File: app/src/main/java/com/example/learingrealmandretrofit/objects/CardRealm.kt [hunk 3]
```diff
@@ -20,4 +20,4 @@ open class CardRealm(
     override fun toString(): String {
         return Gson().toJson(this)
     }
-}
\ No newline at end of file
+}
```
Comment (skliarov) [inline app/src/main/java/com/example/learingrealmandretrofit/objects/CardRealm.kt:23]: It should be called `Card`.

File: app/src/main/java/com/example/learingrealmandretrofit/objects/DeckRealm.kt [hunk 4]
```diff
@@ -15,4 +15,4 @@ open class DeckRealm(
     override fun toString(): String {
         return Gson().toJson(this)
     }
-}
\ No newline at end of file
+}
```
Comment (skliarov) [inline app/src/main/java/com/example/learingrealmandretrofit/objects/DeckRealm.kt:18]: It should be called `Deck`.

File: app/src/main/res/layout/card_fragment_recycler.xml [hunk 5]
```diff
@@ -31,7 +31,7 @@
             tools:listitem="@layout/item_card_recycler" />
 
         <com.google.android.material.floatingactionbutton.FloatingActionButton
-            android:id="@+id/buttonFloatingAction"
+            android:id="@+id/floatingActionButtonCard"
```
Comment (skliarov) [inline app/src/main/res/layout/card_fragment_recycler.xml:34]: This ID says nothing about what it's for.

File: app/src/main/res/navigation/authentication_navigation.xml [hunk 6]
```diff
@@ -7,7 +7,7 @@
     <fragment
         android:id="@+id/signUpFragment"
         android:name="com.example.learingrealmandretrofit.fragment.SignUpFragment"
-        android:label="SignUpFragment"
+        android:label="SIGN UP"
```
Comment (skliarov) [inline app/src/main/res/navigation/authentication_navigation.xml:10]: Where this label should be?

File: app/src/main/res/navigation/authentication_navigation.xml [hunk 7]
```diff
@@ -16,7 +16,7 @@
     <fragment
         android:id="@+id/signInFragment"
         android:name="com.example.learingrealmandretrofit.fragment.SignInFragment"
-        android:label="SignInFragment"
+        android:label="SIGN IN"
```
Comment (skliarov) [inline app/src/main/res/navigation/authentication_navigation.xml:19]: Where this label should be?

File: app/src/main/res/navigation/card_navigation.xml [hunk 8]
```diff
@@ -8,7 +8,7 @@
     <fragment
         android:id="@+id/cardFragment"
         android:name="com.example.learingrealmandretrofit.fragment.CardFragment"
-        android:label="CardFragment"
+        android:label="Card"
```
Comment (skliarov) [inline app/src/main/res/navigation/card_navigation.xml:11]: Where this label should be?

File: app/src/main/res/navigation/deck_navigation.xml [hunk 9]
```diff
@@ -8,7 +8,7 @@
     <fragment
         android:id="@+id/deckFragment"
         android:name="com.example.learingrealmandretrofit.fragment.DeckFragment"
-        android:label="DeckFragment"
+        android:label="List"
```
Comment (skliarov) [inline app/src/main/res/navigation/deck_navigation.xml:11]: Where this label should be?

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -12,9 +12,15 @@
...
-    <string name="error_server">Oops an error occurred, please try again later.</string>
+    <string name="connection_issues">Oops an error occurred, please try again later.</string>
     <string name="code_error_message">Error code: %1$d</string>
     <string name="card_deletion">Card deletion</string>
     <string name="delete_card">Delete card</string>
     <string name="remove_from_list">Remove from list</string>
+    <string name="register">Register</string>
+    <string name="have_account">Already have an account?</string>
+    <string name="dont_have_account">Don\'t have an account?</string>
+    <string name="sign_in">Sign in</string>
```
Comment (skliarov) [inline app/src/main/res/values/strings.xml:23]: Sign In

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -12,9 +12,15 @@
...
+    <string name="sign_up">Sign up</string>
```
Comment (skliarov) [inline app/src/main/res/values/strings.xml:24]: Sign Up

File: app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -12,9 +12,15 @@
     <string name="delete_list">Delete list</string>
     <string name="delete_list_with_card">Delete list with card</string>
     <string name="network_error_message">Please check your internet connection</string>
-    <string name="error_server">Oops an error occurred, please try again later.</string>
+    <string name="connection_issues">Oops an error occurred, please try again later.</string>
     <string name="code_error_message">Error code: %1$d</string>
     <string name="card_deletion">Card deletion</string>
     <string name="delete_card">Delete card</string>
     <string name="remove_from_list">Remove from list</string>
+    <string name="register">Register</string>
```
Comment (skliarov) [inline app/src/main/res/values/strings.xml:20]: You have to leave one of: `Register` or `Sign Up`

**Comment by skliarov** [comment]
@osmovia I only see some cleanup, the code that we discussed today is not here.

File: app/src/main/java/com/example/learingrealmandretrofit/objects/DeckRealm.kt (hunk 4 above)
Comment (osmovia) [reply app/src/main/java/com/example/learingrealmandretrofit/objects/DeckRealm.kt:18]: I have model Deck for request. How right to name DeckRealm? 
https://github.com/osmovia/LearingRealmRetrofit/blob/main/app/src/main/java/com/example/learingrealmandretrofit/objects/Deck.kt

File: app/src/main/res/navigation/authentication_navigation.xml (hunk 6 above)
Comment (osmovia) [reply app/src/main/res/navigation/authentication_navigation.xml:10]: In toolbar.
photo_2021-11-27_20-43-51

File: app/src/main/java/com/example/learingrealmandretrofit/objects/DeckRealm.kt (hunk 4 above)
Comment (skliarov) [reply app/src/main/java/com/example/learingrealmandretrofit/objects/DeckRealm.kt:18]: My comment remains the same, you should use `Deck` for Realm objects, the other class should be renamed to something else.

File: app/src/main/res/navigation/authentication_navigation.xml (hunk 6 above)
Comment (skliarov) [reply app/src/main/res/navigation/authentication_navigation.xml:10]: It should be in strings file 🤦

**Pull request merged**